.gitlab-ci.yml

```diff
@@ -0,0 +1,103 @@
+variables:
+  KUBE_NAMESPACE:                  "processbot"
+  CI_REGISTRY:                     "paritytech"
+  DOCKER_TAG:                      "${CI_COMMIT_TAG}"
+  GIT_STRATEGY:                    fetch
+  GIT_DEPTH:                       3
+
+stages:
+  - check
+  - test
+  - dockerize
+  - deploy
+
+.docker-env:                       &docker-env
+  image:                           parity/rust-builder:latest
+  before_script:
+    - rustup show
+    - cargo --version
+    - sccache -s
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+      - unknown_failure
+      - api_failure
+  dependencies:                    []
+  interruptible:                   true
+  tags:
+    - linux-docker
+
+#### stage:                        check
+
+check:
+  stage:                           check
+  <<:                              *docker-env
+  script:
+    - cargo fmt --all -- --check
+
+#### stage:                        test
+
+test:
+  stage:                           test
+  <<:                              *docker-env
+  script:
+    - cargo test --all
+
+#### stage:                        dockerize
+
+.build_and_push:                   &build_and_push
+  image:                           docker:18-git
+  services:
+    - docker:18-dind
+  variables:
+    DOCKER_DRIVER:                 overlay2
+    DOCKER_HOST:                   tcp://localhost:2375
+    DOCKER_IMAGE:                  "${CI_REGISTRY}/${KUBE_NAMESPACE}"
+    DOCKER_IMAGE_FULL_NAME:        "$DOCKER_IMAGE:$DOCKER_TAG"
+  script:
+    - echo "$Docker_Hub_Pass_Parity" | docker login -u "$Docker_Hub_User_Parity" --password-stdin
+    - eval "docker build -t" "$DOCKER_IMAGE_FULL_NAME" .
+    - eval "docker push" "$DOCKER_IMAGE_FULL_NAME"
+  tags:
+    - kubernetes-parity-build
+      
+dockerize-processbot:
+  stage:                           dockerize
+  <<:                              *build_and_push
+  environment:
+    name:                          parity-build
+  only:
+    - tags
+
+#### stage:                        deploy
+
+.deploy-k8s:                       &deploy-k8s
+  image:                           parity/kubetools:helm3
+  script:
+    # check syntax
+    - for f in kubernetes/processbot/*yaml; do echo $f; cat $f | envsubst | kubectl --dry-run --validate -n processbot apply -f -; done
```

Comment:
the `-n` should be the environment variable from L2 then (also below)

Reply:
or the whole envsubst can be left out if there is no environment variable in the yaml

Reply:
and afaik it should also work directly without the for loop (even if there were multiple files)

Reply:
@lovelaced i overlooked this before: the docker tag resp build tag probably wants to be a variable in the yaml.